tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -919,7 +919,7 @@ private static Trace getOrCreateCurrentTrace() {
     @VisibleForTesting
     static void clearCurrentTrace() {
         logClearingTrace();
-        currentTrace.remove();
+        currentTrace.set(null);
         MDC.remove(Tracers.TRACE_ID_KEY);
         MDC.remove(Tracers.TRACE_SAMPLED_KEY);
         MDC.remove(Tracers.REQUEST_ID_KEY);
```

Comment:
I didn't touch these as I didn't see these pop in profiles, and log4j2 uses `org.apache.logging.log4j.spi.GarbageFreeSortedArrayThreadContextMap` & `org.apache.logging.log4j.util.SortedArrayStringMap` where the `MDC.remove` & `MDC.put` don't directly free/allocate